### PR TITLE
Modal for approve transactions explanation

### PR DESCRIPTION
**Pull request by ikadushkina**
Task: https://app.asana.com/0/1201289799176936/1201432263059541
Updated Claim Pack modal: a bit updated claim step by Figma, added approve transactions step
Result:
Снимок экрана 2021-11-30 в 11 42 22

### CODE REVIEW

File: js/packages/web/src/views/pack/components/RedeemModal/components/TransactionApprovalStep/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+interface IPropsTransactionApprovalStep {
+  goBack: () => void,
+}
+
+const TransactionApprovalStep = ({ goBack }: IPropsTransactionApprovalStep) => {
+
+  const transactions = ['Opening your Pack.', 'Begin adding new Cards to your wallet.'];
+
+  return (
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/RedeemModal/components/TransactionApprovalStep/index.tsx:12]: Can be a function without return & curly braces with immediate return

File: js/packages/web/src/views/pack/components/RedeemModal/index.tsx [hunk 2]
```diff
@@ -93,6 +105,22 @@ const RedeemModal = ({
...
   }, [modalState, onClose, setModalState]);
 
+  const onClickOpen = () => {
+    if (modalState === openState.Ready) setModalState(openState.TransactionApproval);
+    else handleOpenPack();
+  };
+
+  useEffect(() => {
+    if (!contentRef.current) return;
+    if (modalState === openState.TransactionApproval) {
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/RedeemModal/index.tsx:115]: What does this function?

File: js/packages/web/src/views/pack/components/RedeemModal/index.tsx [hunk 3]
```diff
@@ -102,50 +130,38 @@ const RedeemModal = ({
       visible={isModalVisible}
       onCancel={handleClose}
       footer={null}
-      closable={modalState !== openState.Finding}
+      closable={isModalClosable}
     >
       <div className="modal-redeem">
-        {modalState === openState.Ready && (
+        {
+          isFirstsSteps ? (
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/RedeemModal/index.tsx:137]: No ternary, please. It's clearer to use 2 expressions.

{isFirstSteps && ()}
{!isFirstSteps && ()}

File: js/packages/web/src/views/pack/components/RedeemModal/index.tsx [hunk 4]
```diff
@@ -102,50 +130,38 @@ const RedeemModal = ({
...
           <>
-            <div>
-              <p className="modal-redeem__title">Claim an NFT</p>
+            <div className="modal-redeem__content-container" ref={contentRef}>
+              <ClaimingPackStep
```
Comment (m-sebastiyan) [inline js/packages/web/src/views/pack/components/RedeemModal/index.tsx:140]: Do you use a scroll left to have an animation here?

A simple if/else for rendering would be a better choice for now.

**Review by m-sebastiyan — CHANGES_REQUESTED** [review]

File: js/packages/web/src/views/pack/components/RedeemModal/index.tsx (hunk 4 above)
Comment (ikadushkina) [reply js/packages/web/src/views/pack/components/RedeemModal/index.tsx:140]: Yes, I use a scroll for animation. I discussed animation on this modal with Bohdan

File: js/packages/web/src/views/pack/components/RedeemModal/index.tsx (hunk 2 above)
Comment (ikadushkina) [reply js/packages/web/src/views/pack/components/RedeemModal/index.tsx:115]: This function applies scrolling depending on step

**Review by m-sebastiyan — APPROVED** [review]

**Pull request merged**